addons/point_of_sale/models/pos_session.py

```diff
@@ -1559,7 +1559,7 @@ def try_cash_in_out(self, _type, amount, reason, extras):
         if not sessions:
             raise UserError(_("There is no cash payment method for this PoS Session"))
 
-        self.env['account.bank.statement.line'].create([
+        self.env['account.bank.statement.line'].sudo().create([
```

Comment:
can you check that the user at least has the `pos_user` rights before doing `sudo` ?

Reply:
The `pos.session` model has it implicit as its only allowed group is already `point_of_sale.group_pos_user`. So this method will fail for anyone else. Should we do that explicit check?